app/src/panels/label/panel-label.vue

```diff
@@ -1,16 +1,99 @@
 <script setup lang="ts">
-withDefaults(
+import { onMounted, onBeforeUnmount, onUpdated, ref } from 'vue';
+import { useAutoFontFit } from '@/composables/use-auto-fit-text';
+
+const props = withDefaults(
 	defineProps<{
 		showHeader?: boolean;
 		text?: string;
+		whiteSpace?: string | undefined;
 		color?: string | undefined;
+		textAlign?: string | undefined;
+		fontSize?: 'sm' | 'md' | 'lg' | 'auto';
+		fontWeight?: number | undefined;
+		fontStyle?: string | undefined;
+		font?: 'sans-serif' | 'serif' | 'monospace';
 	}>(),
 	{
 		showHeader: false,
 		text: '',
 		color: undefined,
+		whiteSpace: 'normal',
+		textAlign: 'center',
+		fontSize: 'auto',
+		fontWeight: 800,
+		fontStyle: undefined,
+		font: 'sans-serif',
 	}
 );
+
+const labelContainer = ref<HTMLDivElement | null>(null);
+const labelText = ref<HTMLParagraphElement | null>(null);
+
+const { adjustFontSize } = useAutoFontFit(labelContainer, labelText);
+
+let resizeObserver: ResizeObserver | null = null;
+
+function adjustPadding() {
+	const container = labelContainer.value;
+	if (!container) return '0px';
+
+	const paddingWidth = container.offsetWidth * 0.05;
+	const paddingHeight = container.offsetHeight * 0.05;
+
+	const padding = Math.round(Math.max(12, Math.min(paddingWidth, paddingHeight)));
+
+	if (props.showHeader == true) {
+		container.style.padding = '0px 12px 12px 12px';
+	} else {
+		container.style.padding = `${padding}px`;
+	}
+
+	return;
+}
+
+function updateFit() {
+	adjustPadding();
+	adjustFontSize();
+
+	if (!resizeObserver) {
+		const container = labelContainer.value;
+		if (!container) return;
+
+		// Create a ResizeObserver to watch for changes in the container's dimensions
+		resizeObserver = new ResizeObserver(() => {
+			updateFit();
+		});
+
+		resizeObserver.observe(container);
+	}
+
+	adjustFontSize();
+}
+
+onMounted(() => {
+	if (props.fontSize == 'auto') {
```

Comment:
Strict equality operator is generally preferred
```suggestion
	if (props.fontSize === 'auto') {
```